src/me/mrCookieSlime/Slimefun/Android/ProgrammableAndroid.java

```diff
@@ -700,36 +706,64 @@ private void move(Block b, BlockFace face, Block block) throws Exception {
 		}
 	}
 
-	private void mine(Block b, Block block) {
+	private void mine(Block b, Block block, SlimefunItem item) {
 		Collection<ItemStack> drops = block.getDrops();
 		if (!blockblacklist.contains(block.getType()) && !drops.isEmpty() && CSCoreLib.getLib().getProtectionManager().canBuild(UUID.fromString(BlockStorage.getBlockInfo(b, "owner")), block)) {
-			ItemStack[] items = drops.toArray(new ItemStack[drops.size()]);
-			if (fits(b, items)) {
-				pushItems(b, items);
-				block.getWorld().playEffect(block.getLocation(), Effect.STEP_SOUND, block.getType());
-				block.setType(Material.AIR);
+			if(item != null) {
+				if(fits(b, item.getItem())) {
+					pushItems(b, BlockStorage.retrieve(block));
+					if(SlimefunItem.blockhandler.containsKey(item.getID())) SlimefunItem.blockhandler.get(item.getID()).onBreak(null, block, item, UnregisterReason.ANDROID_DIG);
```

Comment:
Block Handlers return booleans.
Use them, because it might be that a Slimefun block doesn't want to be broken by an Android.